File: strive/121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min_day = INT_MAX;
        int profit = 0;

        for (int price : prices){
            if (price < min_day){
                min_day = price;
            }
            profit = max(profit,price-min_day);
        }
        return profit;
    }
};

File: strive/121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        lowest_price = float('inf')
        profit = 0
        for price in prices:
            lowest_price = min(lowest_price,price)
            profit = max(profit,price - lowest_price)
        return profit
            